x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/dataframe/transforms/pivot/PivotConfig.java

```diff
@@ -144,4 +151,66 @@ public boolean isValid() {
     public static PivotConfig fromXContent(final XContentParser parser, boolean lenient) throws IOException {
         return lenient ? LENIENT_PARSER.apply(parser, null) : STRICT_PARSER.apply(parser, null);
     }
+
+    /**
+     * Does the following checks:
+     *
+     *  - determines if there are any full duplicate names between the aggregation names and the group by names.
+     *  - finds if there are conflicting name paths that could cause a failure later when the config is started.
+     *
+     * Examples showing conflicting field name paths:
+     *
+     * aggName1: foo.bar.baz
+     * aggName2: foo.bar
+     *
+     * This should fail as aggName1 will cause foo.bar to be an object, causing a conflict with the use of foo.bar in aggName2.
+     *
+     * @return List of validation failure messages
+     */
+    public List<String> aggFieldValidation() {
+        if ((aggregationConfig.isValid() && groups.isValid()) == false) {
+            return Collections.emptyList();
+        }
+
+        List<String> validationFailures = new ArrayList<>();
+        List<String> usedNames = new ArrayList<>();
+        // TODO this will need changed once we allow multi-bucket aggs + field merging
+        aggregationConfig.getAggregatorFactories().forEach(agg -> addAggNames(agg, usedNames));
+        aggregationConfig.getPipelineAggregatorFactories().forEach(agg -> addAggNames(agg, usedNames));
+        usedNames.addAll(groups.getGroups().keySet());
+
+        Set<String> leafNames = new HashSet<>(usedNames.size());
+        for(String name : usedNames) {
+            if (leafNames.contains(name)) {
+                validationFailures.add("duplicate field [" + name + "] detected");
+            }
+            leafNames.add(name);
+        }
+
+        for (String fullName : usedNames) {
+            String[] tokens = fullName.split("\\.");
```

Comment:
you omit the dots, so what if I have `foo.bar.baz` and `foobar`? If I get it right, this would fail validation.

Reply:
You are correct. I need to fix that